frontend/src/components/OdhAppCard.tsx

```diff
@@ -196,18 +196,26 @@ const OdhAppCard: React.FC<OdhAppCardProps> = ({ odhApp }) => {
         }}
         style={{ paddingRight: 0 }}
       >
-        <BrandImage src={odhApp.spec.img} alt={odhApp.spec.displayName} />
+        <BrandImage
+          src={odhApp.spec.img}
+          alt={odhApp.spec.displayName}
+          data-testid={`${odhApp.spec.displayName}-image`}
+        />
       </CardHeader>
       <SupportedAppTitle odhApp={odhApp} />
       <CardBody>
         {!dashboardConfig.spec.dashboardConfig.disableISVBadges &&
         odhApp.spec.category &&
         odhApp.spec.support !== ODH_PRODUCT_NAME ? (
           <div className="odh-card__partner-badge-container">
-            <span className="odh-card__partner-badge">{odhApp.spec.category}</span>
+            <span className="odh-card__partner-badge" data-testid={odhApp.spec.category}>
```

Comment:
Use a static strings for test IDs. You can later assert that the element contains the text you're expecting that is dynamic at runtime.
```suggestion
            <span className="odh-card__partner-badge" data-testid="category">
```

Reply:
may I know the reason to give only static strings to data-testids?

Reply:
The structure of a card is static. It has a title, an image, etc... We should be looking for static elements and then asserting the content is of an expected value.

Using scoped selectors and static strings will minimize inadvertently having IDs that overlap.